src/editor/window.rs

```diff
@@ -195,23 +196,56 @@ impl Window {
             }
 
             width += 1;
-            // The previous character is double width, so send this as its own draw command.
-            if character.is_empty() {
-                break;
+
+            // We can't deal with clusters that are longer than 255 bytes, so replace them with spaces.
+            // This should only happen if Neovim sends corrupted lines, or maybe in some pathological
+            // Unicode combining sequence.
+            let cluster = if cluster.len() > 255 { " " } else { cluster };
```

Comment:
is this a circumstance we should notify the user for? Or at least log?

Reply:
We have the raw grid information in the log, so it should be possible to detect the problem from that. And the user would report that some character is showing a space instead of the actual character. So, I don't think this is a problem.

Realistically this can only happen with [Zalgo](https://en.wikipedia.org/wiki/Zalgo_text) text, but we couldn't render that properly anyway when being restricted to one line.